all_fruits.py

```diff
@@ -0,0 +1,163 @@
+import random, pygame, sys, math
+from pygame.locals import *
+
+white = (255, 255, 255)
+green = (0, 255, 0)
+
+class Fruit(pygame.sprite.Sprite):
+    def __init__(self, screen, image_name):
+        # Call the parent class (Sprite) constructor
+        pygame.sprite.Sprite.__init__(self)
+        self.speed = .3
+        self.screen = screen
+        self.total_time = 0
+        self.time = 0
+
+        self.image =  pygame.image.load(image_name).convert_alpha()
+        self.image.set_colorkey(white)
+        self.image.convert_alpha()
+        self.rect = self.image.get_rect()
+
+        self.x = random.randint(0, self.screen.get_width()-int(self.image.get_width()/2))
+        self.y = self.screen.get_height()+self.image.get_width()
+
+        self.rect.centerx = self.x + (self.image.get_width()/2)
+        self.rect.centery = self.y + (self.image.get_height()/2)
+
+        #figure out direction to fruit to move based on spawn position (left or right)
+        border = screen.get_width()/2
+        #if starts to the left, move over right
+        if (self.x < border):
+            self.direction = 1
+        #if starts over on the right, move left
+        else:
+            self.direction = -1
+
+        #determine realistic trajectory angle that will clear minimum height and not exceed maximum
+        min_height = .6*screen.get_height()
+        max_height = .9*screen.get_height()
+
+        start_distance_from_border = abs(self.x - border)
+        distance_from_border = random.randint(0, int(start_distance_from_border))
+
+        try:
+            min_angle = math.atan(min_height/distance_from_border)
+            max_angle = math.atan(max_height/distance_from_border)
+
+            self.angle = random.randint(int(min_angle), int(max_angle))
+        except:
+            self.angle = math.pi/2
+
+    def move(self, x, y):
+        self.x = x
+        self.y = y
+        self.rect.centerx = x + (self.image.get_width()/2)
+        self.rect.centery = y + (self.image.get_height()/2)
+
+    def toss(self, time_passed):
+        #self.y -= self.speed*time_passed
+        self.total_time = self.total_time+time_passed
+        self.time = self.total_time/800
+
+        x_speed = random.randint(0,25)
+        add_x = math.sin(self.time) * x_speed
+        self.x += self.direction*add_x
+
+        y_speed = random.randint(30,43)
+        add_y = math.cos(self.time) *y_speed
+        self.y -= add_y
+
+    def draw(self):
+        self.screen.blit(self.image, (self.x, self.y))
+
+    def checkCollision(self, other):
+        # returns True or False if apple has collided with other object
```

Comment:
Since this comment describes the specification of the method, it could actually be a docstring. Advantages: signals to the reader that this describes what the code does, not how it works; will show up in generated documentation.